File: ruby/resistor-color-duo/resistor_color_duo.rb
module ResistorColorDuo
  VALUES_BY_COLOR = {
    black: "0",
    brown: "1",
    red: "2",
    orange: "3",
    yellow: "4",
    green: "5",
    blue: "6",
    violet: "7",
    grey: "8",
    white: "9"
  }.freeze

  def self.value(colors)
    colors.first(2).map { |color| VALUES_BY_COLOR[color.to_sym] }.join.to_i
  end
end

File: typescript/resistor-color-duo/resistor-color-duo.ts
export function decodedValue(colors: string[]) {
  const [first, second] = colors
  return COLORS.indexOf(first) * 10 + COLORS.indexOf(second)
}

export const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
]
